src/rust/mangaworld/template/src/template.rs

```diff
@@ -1,17 +1,59 @@
+use alloc::string::ToString;
 use aidoku::{
 	error::Result,
 	helpers::substring::Substring,
 	prelude::*,
 	std::{
 		net::{HttpMethod, Request},
+		html::Node,
 		String, Vec,
 	},
 	Chapter, DeepLink, Filter, FilterType, Manga, MangaContentRating, MangaPageResult, MangaViewer,
 	Page,
 };
-
 use crate::helper::{get_chapter_number, get_search_url, manga_status};
 
+static COOKIE_NAME: &str = "MWCookie=";
+static mut STORED_COOKIE: Option<String> = None;
+
+fn get_html_with_cookie(url: String, referer: Option<&str>) -> Result<Node> {
```

Comment:
don't think this needed to be an owned string, but whatever it's fine
```suggestion
fn get_html_with_cookie(url: &str, referer: Option<&str>) -> Result<Node> {
```